### ISSUE TITLE
Emulator fails batchWrite with: `7 PERMISSION_DENIED: Batch writes require admin authentication`

### ISSUE THREAD

**Issue by edeustace** [firebase/firebase-tools#3833]
I'm writing a js client for for firestore and while trying to test `batchWrite` I get the following error in the emulator:
`7 PERMISSION_DENIED: Batch writes require admin authentication`

I don't get this error when running against the real firestore.

### Environment info

**firebase-tools:** 9.19.0

**Platform:** Linux/Pop_Os 21.04

### Test case + Steps to reproduce

1. run `firebase emulators:start --only firestore --project pape-test --debug`
2. download bundled js: [btest.bundle.js](https://gist.github.com/edeustace/16cc552f6cb108e64fb31a61cb1476fe)

3. in another shell:
`export FIRESTORE_EMULATOR_HOST="0.0.0.0:8080"`
`node btest.bundle.js`

Here is the non-bundled src:

```javascript
const { GoogleAuth } = require('google-auth-library');
const { AuthClient } = require('google-auth-library/build/src/auth/authclient');
const { FirestoreClient } = require('@pape/fs-client');
const { Metadata, credentials, ChannelCredentials, StatusObject } = require('@grpc/grpc-js');

const auth = new GoogleAuth({
  scopes: ['https://www.googleapis.com/auth/cloud-platform', 'https://www.googleapis.com/auth/datastore'],
});

let projectId;

let authClient;
let client;

let cred;

const server = process.env.FIRESTORE_EMULATOR_HOST || 'firestore.googleapis.com';

export const getDbRoot = async () => {
  const projectId = await getProjectId();
  //return `projects/${projectId}/databases/(default)/documents`;
  return `projects/${projectId}/databases/(default)`;
};

const getProjectId = async () => {
  if (!projectId) {
    projectId = await auth.getProjectId();
  }
  console.log('projectId:', projectId);
  return projectId;
};

const getCredentials = async (server) => {
  if (!cred) {
    if (server === 'localhost:8080' || server === '0.0.0.0:8080') {
      cred = credentials.createInsecure();
    } else {
      if (!authClient) {
        authClient = await auth.getClient();
      }
      if (!projectId) {
        projectId = await auth.getProjectId();
      }
      cred = credentials.combineChannelCredentials(
        credentials.createSsl(),
        credentials.createFromGoogleCredential(authClient)
      );
    }
  }
  return cred;
};

const { BatchWriteRequest } = require('@pape/fs-client');

const getClient = async () => {
  if (!cred) {
    cred = await getCredentials(server);
  }

  if (!client) {
    console.log('credentials:', credentials);
    client = new FirestoreClient(server, cred);
  }
  return client;
};
const run = async () => {
  const dbRoot = await getDbRoot();
  const c = await getClient();

  const database = dbRoot.replace('/documents', '');

  const docs = [];
  const req = BatchWriteRequest.fromJSON({
    database,
    writes: docs.map((d) => {
      const root = documentPath ? `${dbRoot}/documents/${documentPath}` : `${dbRoot}/documents`;
      const name = `${root}/${collectionId}/${d.id || autoId()}`;
      console.log('name:', name);
      return {
        update: {
          name,
          fields: encodeFields(d),
        },
      };
    }),
  });
  const m = new Metadata();
  return new Promise((resolve, reject) => {
    c.batchWrite(req, m, (err, res) => {
      console.timeEnd('batchWrite');
      if (err) {
        reject(err);
      } else {
        const joined = res.status.map((s, index) => {
          return { doc: docs[index], status: s };
        });
        const errors = joined.filter((ds) => ds.status.code !== Status.OK);

        if (errors.length > 0) {
          reject(new Error(`Failed  for: ${errors.map((d) => d.doc)}`));
        } else {
          resolve('ok');
        }
      }
    });
  });
};
run()
  .then((result) => console.log(result))
  .catch((e) => console.error(e));

```

### Expected behavior

The batchWrite does not return an error

### Actual behavior
The batchWrite returns: `7 PERMISSION_DENIED: Batch writes require admin authentication`

log output:

```shell
[2021-10-15T12:52:56.883Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
i  emulators: Starting emulators: firestore {"metadata":{"emulator":{"name":"hub"},"message":"Starting emulators: firestore"}}
[2021-10-15T12:52:57.013Z] [hub] writing locator at /tmp/hub-pape-test.json
⚠  firestore: Did not find a Cloud Firestore rules file specified in a firebase.json config file. {"metadata":{"emulator":{"name":"firestore"},"message":"Did not find a Cloud Firestore rules file specified in a firebase.json config file."}}
⚠  firestore: The emulator will default to allowing all reads and writes. Learn more about this option: https://firebase.google.com/docs/emulator-suite/install_and_configure#security_rules_configuration. {"metadata":{"emulator":{"name":"firestore"},"message":"The emulator will default to allowing all reads and writes. Learn more about this option: https://firebase.google.com/docs/emulator-suite/install_and_configure#security_rules_configuration."}}
[2021-10-15T12:52:57.020Z] Ignoring unsupported arg: projectId {"metadata":{"emulator":{"name":"firestore"},"message":"Ignoring unsupported arg: projectId"}}
[2021-10-15T12:52:57.020Z] Ignoring unsupported arg: auto_download {"metadata":{"emulator":{"name":"firestore"},"message":"Ignoring unsupported arg: auto_download"}}
[2021-10-15T12:52:57.020Z] Starting Firestore Emulator with command {"binary":"java","args":["-Dgoogle.cloud_firestore.debug_log_level=FINE","-Duser.language=en","-jar","/home/ede/.cache/firebase/emulators/cloud-firestore-emulator-v1.13.1.jar","--host","0.0.0.0","--port",8080],"optionalArgs":["port","webchannel_port","host","rules","functions_emulator","seed_from_export"],"joinArgs":false} {"metadata":{"emulator":{"name":"firestore"},"message":"Starting Firestore Emulator with command {\"binary\":\"java\",\"args\":[\"-Dgoogle.cloud_firestore.debug_log_level=FINE\",\"-Duser.language=en\",\"-jar\",\"/home/ede/.cache/firebase/emulators/cloud-firestore-emulator-v1.13.1.jar\",\"--host\",\"0.0.0.0\",\"--port\",8080],\"optionalArgs\":[\"port\",\"webchannel_port\",\"host\",\"rules\",\"functions_emulator\",\"seed_from_export\"],\"joinArgs\":false}"}}
i  firestore: Firestore Emulator logging to firestore-debug.log {"metadata":{"emulator":{"name":"firestore"},"message":"Firestore Emulator logging to \u001b[1mfirestore-debug.log\u001b[22m"}}
[2021-10-15T12:52:57.970Z] Oct 15, 2021 1:52:57 PM com.google.cloud.datastore.emulator.firestore.websocket.WebSocketServer start
INFO: Started WebSocket server on ws://0.0.0.0:36511
 {"metadata":{"emulator":{"name":"firestore"},"message":"Oct 15, 2021 1:52:57 PM com.google.cloud.datastore.emulator.firestore.websocket.WebSocketServer start\nINFO: Started WebSocket server on ws://0.0.0.0:36511\n"}}
[2021-10-15T12:52:58.012Z] API endpoint: http://0.0.0.0:8080
 {"metadata":{"emulator":{"name":"firestore"},"message":"API endpoint: http://0.0.0.0:8080\n"}}
[2021-10-15T12:52:58.013Z] If you are using a library that supports the FIRESTORE_EMULATOR_HOST environment variable, run:

   export FIRESTORE_EMULATOR_HOST=0.0.0.0:8080

Dev App Server is now running.

 {"metadata":{"emulator":{"name":"firestore"},"message":"If you are using a library that supports the FIRESTORE_EMULATOR_HOST environment variable, run:\n\n   export FIRESTORE_EMULATOR_HOST=0.0.0.0:8080\n\nDev App Server is now running.\n\n"}}
[2021-10-15T12:52:58.040Z] Ignoring unsupported arg: auto_download {"metadata":{"emulator":{"name":"ui"},"message":"Ignoring unsupported arg: auto_download"}}
[2021-10-15T12:52:58.041Z] Ignoring unsupported arg: port {"metadata":{"emulator":{"name":"ui"},"message":"Ignoring unsupported arg: port"}}
[2021-10-15T12:52:58.041Z] Starting Emulator UI with command {"binary":"node","args":["/home/ede/.cache/firebase/emulators/ui-v1.6.3/server.bundle.js"],"optionalArgs":[],"joinArgs":false} {"metadata":{"emulator":{"name":"ui"},"message":"Starting Emulator UI with command {\"binary\":\"node\",\"args\":[\"/home/ede/.cache/firebase/emulators/ui-v1.6.3/server.bundle.js\"],\"optionalArgs\":[],\"joinArgs\":false}"}}
i  ui: Emulator UI logging to ui-debug.log {"metadata":{"emulator":{"name":"ui"},"message":"Emulator UI logging to \u001b[1mui-debug.log\u001b[22m"}}
[2021-10-15T12:52:58.106Z] Web / API server started at 0.0.0.0:5000
 {"metadata":{"emulator":{"name":"ui"},"message":"Web / API server started at 0.0.0.0:5000\n"}}

┌─────────────────────────────────────────────────────────────┐
│ ✔  All emulators ready! It is now safe to connect your app. │
│ i  View Emulator UI at http://0.0.0.0:5000                  │
└─────────────────────────────────────────────────────────────┘

┌───────────┬──────────────┬───────────────────────────────┐
│ Emulator  │ Host:Port    │ View in Emulator UI           │
├───────────┼──────────────┼───────────────────────────────┤
│ Firestore │ 0.0.0.0:8080 │ http://0.0.0.0:5000/firestore │
└───────────┴──────────────┴───────────────────────────────┘
  Emulator Hub running at localhost:4400
  Other reserved ports: 4500

Issues? Report them at https://github.com/firebase/firebase-tools/issues and attach the *-debug.log files.
 
[2021-10-15T12:52:59.385Z] Oct 15, 2021 1:52:59 PM io.gapi.emulators.netty.HttpVersionRoutingHandler channelRead
INFO: Detected HTTP/2 connection.
 {"metadata":{"emulator":{"name":"firestore"},"message":"Oct 15, 2021 1:52:59 PM io.gapi.emulators.netty.HttpVersionRoutingHandler channelRead\nINFO: Detected HTTP/2 connection.\n"}}
[2021-10-15T12:52:59.385Z] Oct 15, 2021 1:52:59 PM io.gapi.emulators.netty.HttpVersionRoutingHandler channelRead
INFO: Detected HTTP/2 connection.
 {"metadata":{"emulator":{"name":"firestore"},"message":"Oct 15, 2021 1:52:59 PM io.gapi.emulators.netty.HttpVersionRoutingHandler channelRead\nINFO: Detected HTTP/2 connection.\n"}}
[2021-10-15T12:52:59.827Z] Oct 15, 2021 1:52:59 PM com.google.cloud.datastore.emulator.impl.util.WrappedStreamObserver onError
INFO: operation failed: Batch writes require admin authentication.
com.google.cloud.datastore.core.exception.DatastoreException: Batch writes require admin authentication.
	at com.google.cloud.datastore.core.exception.DatastoreException$Builder.build(DatastoreException.java:111)
	at com.google.cloud.datastore.emulator.impl.rules.EmulatorRulesAuthorizer.<clinit>(EmulatorRulesAuthorizer.java:70)
	at com.google.cloud.datastore.emulator.impl.rules.EmulatorRulesCompiler.makeAuthorizer(EmulatorRulesCompiler.java:42)
	at com.google.cloud.datastore.emulator.impl.CloudFirestoreV1.<init>(CloudFirestoreV1.java:202)
	at com.google.cloud.datastore.emulator.impl.CloudFirestoreV1Router.lambda$getOrCreateBackend$0(CloudFirestoreV1Router.java:240)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at com.google.cloud.datastore.emulator.impl.CloudFirestoreV1Router.getOrCreateBackend(CloudFirestoreV1Router.java:239)
	at com.google.cloud.datastore.emulator.impl.CloudFirestoreV1Router.batchWrite(CloudFirestoreV1Router.java:179)
	at com.google.cloud.datastore.emulator.firestore.v1.FirestoreV1GrpcAdapter$1.lambda$batchWrite$12(FirestoreV1GrpcAdapter.java:197)
	at com.google.cloud.datastore.emulator.firestore.v1.FirestoreV1GrpcAdapter.unary(FirestoreV1GrpcAdapter.java:67)
	at com.google.cloud.datastore.emulator.firestore.v1.FirestoreV1GrpcAdapter.access$000(FirestoreV1GrpcAdapter.java:39)
	at com.google.cloud.datastore.emulator.firestore.v1.FirestoreV1GrpcAdapter$1.batchWrite(FirestoreV1GrpcAdapter.java:197)
	at com.google.firestore.v1.FirestoreGrpc$MethodHandlers.invoke(FirestoreGrpc.java:1244)
	at io.grpc.stub.ServerCalls$UnaryServerCallHandler$UnaryServerCallListener.onHalfClose(ServerCalls.java:182)
	at io.grpc.PartialForwardingServerCallListener.onHalfClose(PartialForwardingServerCallListener.java:35)
	at io.grpc.ForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:23)
	at io.grpc.ForwardingServerCallListener$SimpleForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:40)
	at io.grpc.Contexts$ContextualizedServerCallListener.onHalfClose(Contexts.java:86)
	at io.grpc.PartialForwardingServerCallListener.onHalfClose(PartialForwardingServerCallListener.java:35)
	at io.grpc.ForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:23)
	at io.grpc.ForwardingServerCallListener$SimpleForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:40)
	at io.grpc.Contexts$ContextualizedServerCallListener.onHalfClose(Contexts.java:86)
	at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.halfClosed(ServerCallImpl.java:331)
	at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed.runInContext(ServerImpl.java:866)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

 {"metadata":{"emulator":{"name":"firestore"},"message":"Oct 15, 2021 1:52:59 PM com.google.cloud.datastore.emulator.impl.util.WrappedStreamObserver onError\nINFO: operation failed: Batch writes require admin authentication.\ncom.google.cloud.datastore.core.exception.DatastoreException: Batch writes require admin authentication.\n\tat com.google.cloud.datastore.core.exception.DatastoreException$Builder.build(DatastoreException.java:111)\n\tat com.google.cloud.datastore.emulator.impl.rules.EmulatorRulesAuthorizer.<clinit>(EmulatorRulesAuthorizer.java:70)\n\tat com.google.cloud.datastore.emulator.impl.rules.EmulatorRulesCompiler.makeAuthorizer(EmulatorRulesCompiler.java:42)\n\tat com.google.cloud.datastore.emulator.impl.CloudFirestoreV1.<init>(CloudFirestoreV1.java:202)\n\tat com.google.cloud.datastore.emulator.impl.CloudFirestoreV1Router.lambda$getOrCreateBackend$0(CloudFirestoreV1Router.java:240)\n\tat java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)\n\tat com.google.cloud.datastore.emulator.impl.CloudFirestoreV1Router.getOrCreateBackend(CloudFirestoreV1Router.java:239)\n\tat com.google.cloud.datastore.emulator.impl.CloudFirestoreV1Router.batchWrite(CloudFirestoreV1Router.java:179)\n\tat com.google.cloud.datastore.emulator.firestore.v1.FirestoreV1GrpcAdapter$1.lambda$batchWrite$12(FirestoreV1GrpcAdapter.java:197)\n\tat com.google.cloud.datastore.emulator.firestore.v1.FirestoreV1GrpcAdapter.unary(FirestoreV1GrpcAdapter.java:67)\n\tat com.google.cloud.datastore.emulator.firestore.v1.FirestoreV1GrpcAdapter.access$000(FirestoreV1GrpcAdapter.java:39)\n\tat com.google.cloud.datastore.emulator.firestore.v1.FirestoreV1GrpcAdapter$1.batchWrite(FirestoreV1GrpcAdapter.java:197)\n\tat com.google.firestore.v1.FirestoreGrpc$MethodHandlers.invoke(FirestoreGrpc.java:1244)\n\tat io.grpc.stub.ServerCalls$UnaryServerCallHandler$UnaryServerCallListener.onHalfClose(ServerCalls.java:182)\n\tat io.grpc.PartialForwardingServerCallListener.onHalfClose(PartialForwardingServerCallListener.java:35)\n\tat io.grpc.ForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:23)\n\tat io.grpc.ForwardingServerCallListener$SimpleForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:40)\n\tat io.grpc.Contexts$ContextualizedServerCallListener.onHalfClose(Contexts.java:86)\n\tat io.grpc.PartialForwardingServerCallListener.onHalfClose(PartialForwardingServerCallListener.java:35)\n\tat io.grpc.ForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:23)\n\tat io.grpc.ForwardingServerCallListener$SimpleForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:40)\n\tat io.grpc.Contexts$ContextualizedServerCallListener.onHalfClose(Contexts.java:86)\n\tat io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.halfClosed(ServerCallImpl.java:331)\n\tat io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed.runInContext(ServerImpl.java:866)\n\tat io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)\n\tat io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\n\n"}}
[2021-10-15T12:53:02.383Z] Received signal SIGINT (Ctrl-C) 1
 
i  emulators: Received SIGINT (Ctrl-C) for the first time. Starting a clean shutdown. 
i  emulators: Please wait for a clean shutdown or send the SIGINT (Ctrl-C) signal again to stop right now. 
i  emulators: Shutting down emulators. {"metadata":{"emulator":{"name":"hub"},"message":"Shutting down emulators."}}
i  ui: Stopping Emulator UI {"metadata":{"emulator":{"name":"ui"},"message":"Stopping Emulator UI"}}
⚠  Emulator UI has exited upon receiving signal: SIGINT 
i  firestore: Stopping Firestore Emulator {"metadata":{"emulator":{"name":"firestore"},"message":"Stopping Firestore Emulator"}}
[2021-10-15T12:53:02.395Z] *** shutting down gRPC server since JVM is shutting down
 {"metadata":{"emulator":{"name":"firestore"},"message":"*** shutting down gRPC server since JVM is shutting down\n"}}
[2021-10-15T12:53:02.403Z] *** server shut down
 {"metadata":{"emulator":{"name":"firestore"},"message":"*** server shut down\n"}}
i  hub: Stopping emulator hub {"metadata":{"emulator":{"name":"hub"},"message":"Stopping emulator hub"}}
i  logging: Stopping Logging Emulator {"metadata":{"emulator":{"name":"logging"},"message":"Stopping Logging Emulator"}}

```

**Comment by jwojnowski** [comment]
I know this is a pretty old issue, but finding an answer isn't that easy, so I'll leave a note anyway.

After spending way too much time on it, I stumbled upon [a similar issue](https://github.com/firebase/firebase-tools/issues/1363). I tried [this answer](https://github.com/firebase/firebase-tools/issues/1363#issuecomment-498364771) and it worked for me with a `batchWrite`.

It looks like the solution is as simple as using `owner` as your token:
```
Authorization: Bearer owner
```
